### PR TITLE
Braedon

### PR DESCRIPTION
**Pull request by braedons**
**Summary**
The state `addedCourse` was created for SectionTableBody. It becomes true when the add button is clicked in a class search, or upon creation for the row, if the course is already in the list of added ones.

The html class `addedCourse` was also created, and is added to the row if the state `addedCourse` is true.

The function to set the state was also added to the objects in the addedCourses array to allow for removing highlights.

image
image

**Test Plan**
- Verify that clicking add highlights the corresponding row
- Verify that deleting a course from the schedule removes the highlight
- Verify that all courses are highlighted in the Added Classes tab
- Verify that switching schedule numbers highlights only the corresponding courses
- Verify that doing a new search retains highlights

**Issues**
#26

### CODE REVIEW

File: client/src/components/SectionTable/SectionTableBody.js [hunk 1]
```diff
@@ -376,12 +380,41 @@ const StatusCell = withStyles(styles)((props) => {
...
+                setAddedCourse(true);
+            }
+        });
+    }
+
+    let addedCourseStr = addedCourse ? ' addedCourse' : '';
+
     return (
-        <tr className={classes.tr}>
+        <tr className={classes.tr + addedCourseStr}>
```
Comment (ChaseC99) [inline client/src/components/SectionTable/SectionTableBody.js:404]: We should probably use [classnames](https://github.com/JedWatson/classnames) here. 

The library is already in the project, so we can just import it at the top
```javascript
import classNames from 'classnames';
```

And use it in the return
```javascript
return (
    <tr className={classNames(classes.tr, {"addedCourse": addedCourse})}>
```

**Review by ChaseC99 — CHANGES_REQUESTED** [review]
Great work on this! Looks awesome and seems like it works perfectly. Even works when you switch schedules 👏 

I have a few comments but once those are resolved, it should be good to go.

**Comment by ChaseC99** [comment]
I missed this in my first Code Review, but it looks like an older version of the backend submodule was also accidentally committed.

Try running `git submodule update` and commit the change to bring it back up to date.

**Comment by braedons** [comment]
I ran that command, but there was no output. `git status` also shows no changes.

File: client/src/components/Calendar/calendar.css [hunk 2]
```diff
@@ -1,5 +1,9 @@
 .rbc-header {
     border-bottom: none;
+
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
```
Comment (ChaseC99) [inline client/src/components/Calendar/calendar.css:6]: LGTM!

**Comment by devsdevsdevs** [comment]
First, sorry for the massive massive delay on this review. 
I was trying to think of a better way to implement this change as I didn't really feel that we needed to pass the state modifying function around to make the state changes. 
Instead, I implemented a lookup table into AppStore and the `SectionTableBody` component responds to changes when courses are added/removed and when the schedule number is changed. This is less confusing and more in line with React patterns. However, this is more computationally expensive as every SectionTableBody will need to check the lookup table when courses are added or the schedule index is changed, but in practice, the performance impact is very negligible. You can check out the changes in the `section-highlight` branch. Nonetheless, thank you again for your effort, description and test plan.

**Comment by devsdevsdevs** [comment]
I've cherry picked and pushed the latest commit from here (a24180b) to develop. Can you please close this branch and address other issues within their own branches?

**Pull request closed without merging**